overlord/hookstate/ctlcmd/helpers.go

```diff
@@ -96,7 +96,19 @@ func runServiceCommand(context *hookstate.Context, inst *servicestate.Instructio
 		return err
 	}
 
-	ts, err := servicestateControl(st, appInfos, inst)
+	isSameChange := func(otherTask *state.Task) bool {
+		if context.IsEphemeral() {
+			return false
+		}
+		task, ok := context.Task()
```

Comment:
no reason to do this prep bits inside the function, they could be outside, ex. if context is ephemeral don't pass a function, pass see comment below, this helper probably doesn't belong here (at least until we have a strong case)